### ISSUE TITLE
rssDocument parser fails with InvalidEndElement on a valid rss feed

### ISSUE THREAD

**Issue by TOTBWF** [k0ral/rss-conduit#8]
When trying to parse [this](http://typetheorypodcast.com/feed/podcast) rss feed, the `rssDocument` parser fails with `InvalidEndElement`. The issue seems to lie in a weird nesting of `category` tags.
```xml
<itunes:category text="Science &amp; Medicine"><itunes:category text="Natural Sciences"></itunes:category></itunes:category>
```

**Comment by k0ral** [comment]
It so happens that `<category>` is part of both the RSS2 standard, and the iTunes podcast extension. The only way to disambiguate them is through proper namespace handling, but right now `rss-conduit` is too lenient with XML namespaces: it simply ignores them by default, hence the issue.
From the RSS2 specification:
> The elements defined in this document are not themselves members of a namespace
I will therefore restrict RSS2 parsers to elements with no namespace.

**Comment by k0ral** [comment]
Fixed in 81c0195d2f1c6cad2e30704c2176a5964e9dc277.
Released as [0.4.3.1](https://hackage.haskell.org/package/rss-conduit-0.4.3.1).

**Comment by TOTBWF** [comment]
Awesome, thank you so much :smile:
